src/objects/api/utils.py

```diff
@@ -43,3 +43,22 @@ def display_choice_values_for_help_text(choices: DjangoChoices) -> str:
         items.append(item)
 
     return "\n".join(items)
+
+
+def merge_patch(target: Any, patch: Any) -> Dict[str, Any]:
+    """An implementation of https://datatracker.ietf.org/doc/html/rfc7396 - JSON Merge Patch"""
+    if not isinstance(patch, dict):
+        return patch
+
+    if not isinstance(target, dict):
+        # Ignore the contents and set it to an empty dict
+        target = {}
+    for k, v in patch.items():
+        if v is None:
+            if k in target:
+                # remove the key/value pair from target
+                del target[k]
```

Comment:
By looking at the tests, I see that this is the desired behaviour, but I wonder about the case where we want to set a key to a null value (for example if the key is required but can take null values) :thinking:

Reply:
This is following the RFC implementation, but I agree when writing this I thought about this use case. So we could maybe deviate from the spec, depending on which use case we cant to be able to achieve